src/pages/plugins/plugins.js

```diff
@@ -285,61 +350,206 @@ export default function PluginsInclude(updates) {
     }
   }
 
-  async function getFilteredPlugins(filterName) {
+  async function getFilteredPlugins(filterState, isInitial = false) {
+    if (!filterState) return;
     if (isLoading || !hasMore) return;
-    
+
     try {
       isLoading = true;
       $list.all.setAttribute("empty-msg", strings["loading..."]);
-      
-      let response;
-      if (filterName === "top_rated") {
-        response = await fetch(`${constants.API_BASE}/plugins?explore=random&page=${currentPage}&limit=${LIMIT}`);
-      } else {
-        response = await fetch(
-          `${constants.API_BASE}/plugin?orderBy=${filterName}&page=${currentPage}&limit=${LIMIT}`,
-        );
-      }
-      const fetchedPlugins = await response.json();
-      
-      if (fetchedPlugins.length < LIMIT) {
-        hasMore = false;
+
+      const { items, hasMore: hasMoreResults } = await retrieveFilteredPlugins(filterState);
+
+      if (currentFilter !== filterState) {
+        return;
       }
-      
+
       const installed = await fsOperation(PLUGIN_DIR).lsDir();
       const disabledMap = settings.value.pluginsDisabled || {};
-      
+
       installed.forEach(({ url }) => {
-        const plugin = fetchedPlugins.find(({ id }) => id === Url.basename(url));
+        const plugin = items.find(({ id }) => id === Url.basename(url));
         if (plugin) {
           plugin.installed = true;
           plugin.enabled = disabledMap[plugin.id] !== true;
           plugin.onToggleEnabled = onToggleEnabled;
           plugin.localPlugin = getLocalRes(plugin.id, "plugin.json");
         }
       });
-      
-      // Add plugins to the all plugins array 
-      plugins.all.push(...fetchedPlugins);
-      
+
+      if (isInitial) {
+        $list.all.querySelectorAll(".filter-empty").forEach((el) => el.remove());
+      }
+
+      plugins.all.push(...items);
+
       const fragment = document.createDocumentFragment();
-      fetchedPlugins.forEach((plugin) => {
+      items.forEach((plugin) => {
         fragment.append(<Item {...plugin} updates={updates} />);
       });
-      $list.all.append(fragment);
-      
-      currentPage++;
-      $list.all.setAttribute("empty-msg", strings["no plugins found"]);
+
+      if (fragment.childNodes.length) {
+        $list.all.append(fragment);
+      } else if (isInitial) {
+        $list.all.append(
+          <div className="filter-empty">{strings["no plugins found"] || "No plugins found"}</div>,
+        );
+      }
+
+      hasMore = hasMoreResults;
+      if (!hasMore) {
+        $list.all.setAttribute("empty-msg", strings["no plugins found"]);
+      }
     } catch (error) {
       $list.all.setAttribute("empty-msg", strings["error"]);
-      window.log("error", "Failed to filter plugins:");
-      window.log("error", error);
+      console.error("Failed to filter plugins:", error);
```

Comment:
`window.log` function usage is missing. That function outputs information to the `acode.log` file, Helpful for verbose/debugging purposes

```suggestion
      console.error("Failed to filter plugins:", error);
      window.log("error", "Failed to filter plugins:");
      window.log("error", error);
```